adapters/deeplabv3/model_adapter.py

```diff
@@ -0,0 +1,480 @@
+from dtlpy.utilities.dataset_generators.dataset_generator_torch import DatasetGeneratorTorch
+from dtlpy.utilities.reports import Report, FigOptions, ConfusionMatrix
+from torchvision.transforms.functional import InterpolationMode
+from sklearn.metrics import confusion_matrix
+from torch.utils.data import DataLoader
+import matplotlib.pyplot as plt
+import torchvision.transforms
+import torch.optim as optim
+import torch.nn.functional
+import seaborn as sns
+import numpy as np
+import dtlpy as dl
+import torch.nn
+import logging
+import torch
+import time
+import tqdm
+import cv2
+import os
+
+logger = logging.getLogger('segmentation-models-adapter')
+
+
+@dl.Package.decorators.module(name='model-adapter',
+                              description='Base Model Adapter for Segmentation Models',
+                              init_inputs={'model_entity': dl.Model})
+class ModelAdapter(dl.BaseModelAdapter):
+
+    def load(self, local_path, **kwargs):
+        """ Loads model and populates self.model with a `runnable` model
+            This function is called by load_from_model (download to local and then loads)
+
+        :param local_path: `str` directory path in local FileSystem
+        """
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        weights = self.configuration.get('weights_filename', None)
+        model_name = self.configuration.get('model_name', 'deeplabv3_resnet50')
+        self.model = torch.hub.load('pytorch/vision:v0.10.0', model_name, pretrained=True)
+        if weights is not None:
+            weights_filename = os.path.join(local_path, self.configuration.get('weights_filename'))
+            if os.path.isfile(weights_filename):
+                logger.info("Loading a model from {}".format(local_path))
+                num_classes = len(self.model_entity.id_to_label_map.items())
+                self.model.classifier[-1] = torch.nn.Conv2d(256, num_classes, kernel_size=(1, 1), stride=(1, 1))
+                self.model.load_state_dict(torch.load(weights_filename, map_location=self.device))
+                logger.info("Loaded custom weights {}".format(weights_filename))
+            else:
```

Comment:
If weights are not found, raise error. If they don't configure any weights filename, then it's ok to use default.

Reply:
i'll add it, thanks!